src/iqm/qiskit_iqm/iqm_backend.py

```diff
@@ -16,81 +16,117 @@
 from __future__ import annotations
 
 from abc import ABC
-import re
-from typing import Final, Optional
+from dataclasses import dataclass
+import itertools
+from typing import Final, Optional, Union
 
 from qiskit.circuit import Parameter
 from qiskit.circuit.library import CZGate, IGate, Measure, RGate
 from qiskit.providers import BackendV2
-from qiskit.transpiler import Target
+from qiskit.transpiler import InstructionProperties, Target
 
-from iqm.iqm_client import QuantumArchitectureSpecification, is_directed_instruction, is_multi_qubit_instruction
+from iqm.iqm_client import DynamicQuantumArchitecture, Locus, QuantumArchitectureSpecification
 from iqm.qiskit_iqm.move_gate import MoveGate
 
 IQM_TO_QISKIT_GATE_NAME: Final[dict[str, str]] = {'prx': 'r', 'cz': 'cz'}
 
 
+@dataclass
+class QuantumArchitecture:
+    """A representation of the quantum computer architecture.
+
+    Used to represent data from a DynamicQuantumArchitecture or a QuantumArchitectureSpecification in a unified way.
+    """
+
+    qubits: list[str]
+    computational_resonators: list[str]
+    gate_loci: dict[str, tuple[Locus, ...]]
```

Comment:
How about just using DynamicQuantumArchitecture instead of this one, and adding a function for creating a fake DQA from a static architecture just for FakeBackends?
You could create a single fake implementation for each gate, containing all the loci.

Reply:
And put something like the all-zeros UUID in the UUID field. It's a hack but it should be safe.

Reply:
Yeah, that's another option I was considering. I did it this way because there doesn't seem to be a clean way to indicate a "fake"/"dummy"/"unused" UUID, but I guess the all-zeros one would be ok. For me either way is fine.

Reply:
Removed QuantumArchitecture now and using a "fake" DQA instead.